File: src/loader.ts
import * as examplePlugin from "../plugin/example"

export async function callbackHandler(callback: any) {
    if (callback.post_type == 'message') {
        return pluginRegistry("message", callback)
    }
    if (callback.post_type == 'request') {
        return pluginRegistry("request", callback)
    }
    if (callback.post_type == 'notice') {
        return pluginRegistry("notice", callback)
    }
    if (callback.post_type == 'meta_event') {
        return pluginRegistry("meta_event", callback)
    }
}

async function pluginRegistry(type: string, callback: any) {
    await examplePlugin.handler(type, callback)
}

File: extend/plugin/loader.go
package plugin

import (
	"github.com/Elyart-Network/NyaBot/core/gocqhttp/cqcall"
	"github.com/gin-gonic/gin"
	"log"
)

func GoCqLoader(ctx *gin.Context, function GoCqInvoker) {
	callback := cqcall.Callback(ctx)
	switch callback.PostType {
	case "message":
		function(callback)
	case "request":
		function(callback)
	case "notice":
		function(callback)
	case "meta_event":
		function(callback)
	default:
		log.Println("Unknown PostType: ", callback.PostType)
	}
	return
}
